shell_integration/MacOSX/NextcloudIntegration/NCDesktopClientSocketKit/LocalSocketClient.m

```diff
@@ -326,24 +326,43 @@ - (void)readFromSocket
 
 - (void)processInBuffer
 {
-    NSLog(@"Processing in buffer. In buffer length %li", [_inBuffer length]);
+    NSLog(@"Processing in buffer. In buffer length %li", _inBuffer.length);
+
     UInt8 separator[] = {0xa}; // Byte value for "\n"
-    while(true) {
-        NSRange firstSeparatorIndex = [_inBuffer rangeOfData:[NSData dataWithBytes:separator length:1] options:0 range:NSMakeRange(0, [_inBuffer length])];
-        
-        if(firstSeparatorIndex.location == NSNotFound) {
-            NSLog(@"No separator found. Stopping.");
-            return; // No separator, nope out
-        } else {
-            unsigned char *buffer = [_inBuffer mutableBytes];
-            buffer[firstSeparatorIndex.location] = 0; // Add NULL terminator, so we can use C string methods
-            
-            NSString *newLine = [NSString stringWithUTF8String:[_inBuffer bytes]];
+    NSData * const separatorData = [NSData dataWithBytes:separator length:1];
+
+    while(_inBuffer.length > 0) {
```

Comment:
@claucambra I am not an expert in Objective-C, but, are you sure the `_inBuffer.length` will reach zero at some point? will not we get an infinite loop?

Reply:
No, as at the end of the iteration we do `[_inBuffer replaceBytesInRange:nullTerminatorRange withBytes:NULL length:0];`